File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis.VisualBasic/BoundNewT.cs
using System.Diagnostics;
using Microsoft.CodeAnalysis.VisualBasic.Symbols;

namespace Microsoft.CodeAnalysis.VisualBasic
{
	internal sealed class BoundNewT : BoundObjectCreationExpressionBase
	{
		public BoundNewT(SyntaxNode syntax, BoundObjectInitializerExpressionBase initializerOpt, TypeSymbol type, bool hasErrors = false)
			: base(BoundKind.NewT, syntax, initializerOpt, type, hasErrors || BoundNodeExtensions.NonNullAndHasErrors(initializerOpt))
		{
		}

		[DebuggerStepThrough]
		public override BoundNode Accept(BoundTreeVisitor visitor)
		{
			return visitor.VisitNewT(this);
		}

		public BoundNewT Update(BoundObjectInitializerExpressionBase initializerOpt, TypeSymbol type)
		{
			if (initializerOpt != base.InitializerOpt || (object)type != base.Type)
			{
				BoundNewT boundNewT = new BoundNewT(base.Syntax, initializerOpt, type, base.HasErrors);
				boundNewT.CopyAttributes(this);
				return boundNewT;
			}
			return this;
		}
	}
}

File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis/VisualBasic/BoundNewT.vb
Imports Microsoft.CodeAnalysis
Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
Imports System
Imports System.Diagnostics

Namespace Microsoft.CodeAnalysis.VisualBasic
	Friend NotInheritable Class BoundNewT
		Inherits BoundObjectCreationExpressionBase
		Public Sub New(ByVal syntax As SyntaxNode, ByVal initializerOpt As BoundObjectInitializerExpressionBase, ByVal type As TypeSymbol, Optional ByVal hasErrors As Boolean = False)
			MyBase.New(BoundKind.NewT, syntax, initializerOpt, type, If(hasErrors, True, initializerOpt.NonNullAndHasErrors()))
		End Sub

		<DebuggerStepThrough>
		Public Overrides Function Accept(ByVal visitor As BoundTreeVisitor) As BoundNode
			Return visitor.VisitNewT(Me)
		End Function

		Public Function Update(ByVal initializerOpt As BoundObjectInitializerExpressionBase, ByVal type As TypeSymbol) As Microsoft.CodeAnalysis.VisualBasic.BoundNewT
			Dim boundNewT As Microsoft.CodeAnalysis.VisualBasic.BoundNewT
			If (initializerOpt <> MyBase.InitializerOpt OrElse CObj(type) <> CObj(MyBase.Type)) Then
				Dim boundNewT1 As Microsoft.CodeAnalysis.VisualBasic.BoundNewT = New Microsoft.CodeAnalysis.VisualBasic.BoundNewT(MyBase.Syntax, initializerOpt, type, MyBase.HasErrors)
				boundNewT1.CopyAttributes(Me)
				boundNewT = boundNewT1
			Else
				boundNewT = Me
			End If
			Return boundNewT
		End Function
	End Class
End Namespace